### PR TITLE
DG Documentation

### PR DESCRIPTION
**Pull request by dinhnhobao**
To go through NFR and Glossary

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -1239,18 +1240,21 @@ image::TaskCardDone.png[]
...
+//tag::task-summary-second[]
+|  `*task* done /code MOD_CODE /id ID_NUMBER`
 |  `*task* list`
 |  `*task* module /code MOD_CODE`
-|  `*task* not-done`
-|  `*task* find DESCRIPTION [MORE_DESCRIPTIONS]...`
+|  `*task* undone`
+|  `*task* find DESCRIPTION [MORE_DESCRIPTIONS]...` +
 |  `*task* search [/date DATE] [/month MONTH] [/year YEAR]`
+//end::task-summary-second[]
```
Comment (alushingg) [inline docs/UserGuide.adoc]: I think should sort by alphabetical order

Comment (dinhnhobao) [reply docs/UserGuide.adoc]: I separated my commands and @NNpanpan's for clearer distinction, and also we can quote the corresponding part in the PPP easier. The commands are organised in terms of logic (CRUD -> queries -> slightly complex queries) so it should be okay. For example, "marking a task as done" should come before "view undone tasks", and "creating tasks" should be before "find tasks by description/date," Alphabetical order may make it hard to make sense of the commands.

Comment (NNpanpan) [reply docs/UserGuide.adoc]: Summary isn't for making sense of commands, but for quick look-up.
If user wants to understand commands, they'd go thru the UG sections, not the summary. So you should keep it alphabetically sorted - consistent with other parts too.

Comment (dinhnhobao) [reply docs/UserGuide.adoc]: Yeah makes sense. I have updated it to be in alphabetical order. Thanks for the feedback!

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -543,42 +546,16 @@ The following activity diagram summarizes what happens when a user executes a `*
...
-The following activity diagram summarizes what happens when a user executes a task find command:
-
-.Activity Diagram for `*task* search` Command
-image::TaskSearchActivityDiagram.png[]
-
 //tag::task-edit[]
 ===== Editing a task
-The `task edit` command allows user to edit a task in `ModManager`. The fields that can be edited are: description and time.
+The `task edit` command allows user to edit a task in Mod Manager.
+
```
Comment (NNpanpan) [inline docs/DeveloperGuide.adoc:552]: Why change to less specific?

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -502,13 +502,16 @@ Generating task number is done through static calls to methods of `TaskNumManage
...
+* `*add*` - Adds a task to a `Module` in Mod Manager.
+* `*edit*` - Edits the information of a task in Mod Manager.
+* `*delete*` - Deletes a task from the `Module` and Mod Manager.
+* `*done*` - Marks a task as done in Mod Manager
+* `*list*` - Shows a list of all tasks across all `Module` s in Mod Manager.
+* `*module*` - Shows a list of all tasks for a specified `Module` in Mod Manager.
+* `*undone*` - Shows a list of all undone tasks in Mod Manager.
+* `*find*` - Finds a task in Mod Manager by its description.
+* `*search*`- Searches for tasks that occur on the specified date, month, or year in Mod Manager.
+* `*upcoming*` - Finds upcoming tasks (for tasks with a specified time period) in Mod Manager. `[coming in v2.0]`
```
Comment (NNpanpan) [inline docs/DeveloperGuide.adoc]: Previous PR already make not-bold to be consistent with other sections.
Why bold again?

**Review by NNpanpan — COMMENTED** [review]
Just change a bit in the DG about bold words

File: docs/DeveloperGuide.adoc (hunk 2 above)
Comment (dinhnhobao) [reply docs/DeveloperGuide.adoc:552]: I resolved the merge conflict on GitHub, there was 200+ lines so maybe I lost it somewhere

**Pull request merged**